File: server/routes/items.js
const express = require("express")
const upload = require("../upload")
const Item = require("../models/Item")

const router = express.Router()

// Retrieve all clothing items in database
router.get("/", async (req, res) => {
  try {
    const items = await Item.find({}).exec()
    res.json(items)
  } catch (e) {
    console.log(e)
  }
})

// Add clothing name, image, and tags to database
router.post("/", upload.none(), async (req, res) => {
  const data = JSON.parse(req.body.data)

  try {
    const item = new Item(data)
    await item.save()
    console.log("Added to database")
  } catch (e) {
    console.log(e)
  }
})

module.exports = router

File: models/routers/items.py
from fastapi import APIRouter
from schemas.item import Item
from database import index
from constants import DIMENSIONS

router = APIRouter(prefix="/items", tags=["items"])


# Retrieve all clothing items in database
@router.get("", response_model=list[Item])
def get_items():
    try:
        items = index.query(
            vector=[0] * DIMENSIONS,
            top_k=1000,
            include_values=True,
            include_metadata=True,
        )

        return items.matches
    except Exception as e:
        print(e)


# Add clothing image, description, tags, and embeddings to database
@router.post("", status_code=201)
def add_item_to_database(item: Item):
    # Converts Pydantic Item model into Python dictionary
    item_dict = item.model_dump(exclude={"score"})
    try:
        index.upsert(vectors=[item_dict])
    except Exception as e:
        print(e)
